clippy_lints/src/large_stack_arrays.rs

```diff
@@ -30,13 +30,15 @@ declare_clippy_lint! {
 pub struct LargeStackArrays {
     maximum_allowed_size: u64,
     prev_vec_macro_callsite: Option<Span>,
+    is_in_const_item: bool,
```

Comment:
@GnomedDev Coming in late: `bool` is not sufficient for something like this:

```rust
const A: u8 = {
    const B: () = ();

    let x = [0u8; 20_000_000];
    x[0]
};
```

This will still lint the array assigned to `x`. But if you comment out `const B`, it won't lint, as you intended in this fix. 

If you want to fix this, you can turn this into a counter, increment it when entering an item and decrementing it when exiting it.

This also won't prevent the lint from triggering in `const fn`s.  But not sure if you intended to also prevent linting there.

Reply:
That's annoying as hell. Open an issue and I'll pick it up later today. I was under the assumption that the point of an item was to not be nested and something would hoist the `const B` definition out of the `A` item, but I guess that's a little silly in retrospect.

Reply:
Thanks! I created https://github.com/rust-lang/rust-clippy/issues/13529 and tried, but failed to assign you to it.

Reply:
Opened a PR to fix!